pkgs/development/compilers/cudatoolkit/extension.nix

```diff
@@ -18,8 +18,8 @@ final: prev: let
   # E.g. for cudaPackages_11_8 we use gcc11 with gcc12's libstdc++
   # Cf. https://github.com/NixOS/nixpkgs/pull/218265 for context
   backendStdenv = final.callPackage ./stdenv.nix {
-    nixpkgsStdenv = prev.pkgs.stdenv;
-    nvccCompatibleStdenv = prev.pkgs.buildPackages."${finalVersion.gcc}Stdenv";
+    nixpkgsCompatibleLibstdcxx = prev.pkgs.buildPackages.gcc.cc.lib; # Or is it targetPackages?
+    nvccCompatibleCC = prev.pkgs.buildPackages."${finalVersion.gcc}".cc;
```

Comment:
Oh, I must have forgot to push the last commit where I actually replaced these with `final`s (as well as dropping `buildPackages` after a suggestion from https://matrix.to/#/#cross-compiling:nixos.org - I was going to leave them out until we have actually tried cross-building any of the cuda packages)

Reply:
> as well as dropping buildPackages after a suggestion from https://matrix.to/#/#cross-compiling:nixos.org - I was going to leave them out until we have actually tried cross-building any of the cuda packages

oh hmm

I think using `pkgs.gcc.cc.lib` instead of `buildPackages.gcc.cc.lib` for `nixpkgsCompatibleLibstdcxx` will work (assuming that `hostPlatform` == `targetPlatform` for the current stage; i.e. that we aren't doing a canadian cross compile or something similar) but I _think_ in situations where we are actually cross-compiling this will result in `gcc` being built an extra time (built by the cross toolchain but this time for `gcc` to _run on_ the platform we're cross compiling for instead of on the build platform; since `targetPlatform` is still the same the resulting `libstdc++` should be ~the same as the previous stage's).

I think it's totally fine to leave it as is for now; when you do actually get to testing cross-compiling CUDA packages it should be pretty apparent if ^ actually is the case (an extra gcc build is pretty hard to miss 🙂)

Reply:
Checks out! I'll leave `buildPackages` and add a comment referring to this discussion